src/test_typing_extensions.py

```diff
@@ -7233,12 +7309,13 @@ def test_or(self):
             Alias | "Ref"
 
     def test_getitem(self):
+        T = TypeVar('T')
         ListOrSetT = TypeAliasType("ListOrSetT", Union[List[T], Set[T]], type_params=(T,))
         subscripted = ListOrSetT[int]
         self.assertEqual(get_args(subscripted), (int,))
         self.assertIs(get_origin(subscripted), ListOrSetT)
-        with self.assertRaises(TypeError):
-            subscripted[str]
+        with self.assertRaises(TypeError, msg="not a generic class"):
```

Comment:
```suggestion
        with self.assertRaisesRegex(TypeError, "not a generic class"):
```
`msg=` is the message that gets raised if the test fails, not the expected message.

Reply:
This currently displays a weired behavior, if I do thie minimal example:
```python
from typing import Tuple, TypeVar, Union, List, Set, Tuple
from typing_extensions import TypeVarTuple, Unpack, get_args, TypeAliasType
T = TypeVar('T')
ListOrSetT = TypeAliasType("ListOrSetT", Union[List[T], Set[T]], type_params=(T,))
subscripted = ListOrSetT[int]
subscripted[int]
TypeError: ListOrSetT[int] is not a generic class
```

However, If I run the test code here in 3.10.0/14 this raises "TypeError: There are no type variables left in ListOrSetT[int]".
So far I have not been able to track down where this change comes from.